Mathlib/SetTheory/Ordinal/Nimber.lean

```diff
@@ -357,4 +360,209 @@ theorem add_trichotomy {a b c : Nimber} (h : a + b + c ≠ 0) :
   · rw [← hx'] at hx
     exact Or.inr <| Or.inr hx
 
+theorem two_nsmul (a : Nimber) : 2 • a = 0 := by
+  rw [succ_nsmul, one_nsmul, add_self]
+
+theorem two_zsmul (a : Nimber) : (2 : ℤ) • a = 0 :=
+  two_nsmul a
+
+theorem add_eq_iff_eq_add : a + b = c ↔ a = c + b :=
+  sub_eq_iff_eq_add
+
+theorem eq_add_iff_add_eq : a = b + c ↔ a + c = b :=
+  eq_sub_iff_add_eq
```

Comment:
I think these should be in `Algebra.CharP.Two`, and here instead prove a `CharP 2` instance.

Reply:
I agree. The reason I didn't add a `CharP 2` instance is because the lemma saying it suffices to prove `0 ≠ 1` and `1 + 1 = 0` didn't exist yet. Is it ok if I copy your proof from the other number branch in the meanwhile?

Reply:
Sure